src/wp-includes/rest-api.php

```diff
@@ -1284,12 +1285,36 @@ function rest_validate_value_from_schema( $value, $args, $param = '' ) {
 			return new WP_Error( 'rest_invalid_param', sprintf( __( '%1$s is not of type %2$s.' ), $param, 'object' ) );
 		}
 
+		/*
+		 * @ticket 48818
+		 */
+		if ( isset( $args['properties'] ) && ! empty( $value ) ) {
+			if ( isset( $args['required'] ) ) { // schema version 4
```

Comment:
This should also check that this is an array, otherwise I think we'll run into issues when we recurse. For instance, the following test has a PHP warning.

```php
public function test_nested_v3_required_property() {
	$schema = array(
		'type'       => 'object',
		'required'   => true,
		'properties' => array(
			'propA' => array(
				'type'       => 'object',
				'required'   => true,
				'properties' => array(
					'propB' => array(
						'type'       => 'object',
						'required'   => true,
						'properties' => array(
							'propC' => array(
								'type'     => 'string',
								'required' => true,
							),
						),
					),
				),
			),
		),
	);

	$value = array(
		'propA' => array(
			'propB' => array(),
		),
	);

	$this->assertWPError( rest_validate_value_from_schema( $value, $schema ) );
}
```

Reply:
done